activesupport/lib/active_support/cache.rb

```diff
@@ -596,9 +596,13 @@ def delete_entry(key, options)
         # Merges the default options with ones specific to a method call.
         def merged_options(call_options)
           if call_options
-            options.merge(call_options)
+            if options.empty?
+              call_options
+            else
+              options.merge(call_options)
+            end
           else
-            options.dup
+            options
```

Comment:
Would not this cause problems if people mutate the return of this method? I guess we should :shipit: and see.

Reply:
I’m out on vacation for the week.

You were right about the empty options case. When they were equal options was actually empty. Your alternative handled more options.

We aren’t mutating anywhere in the Rails codebase and technically this is a private method, however due to inheritance people can still technically use it. We could change the name and introduce a new method or just ship this.

If you need changes I don’t get back until October 2nd ish.

Reply:
Also fwiw I think two guard clauses here makes the code more readable and shorter but I know they’re not preferred.